Search Spots by bounds

TNCodeMonkey opened the pull request:
I've added some functionality to search Spots by bounds, found it really useful in an application I was working on.  I am very new to open source work, so please let me know if I've missed anything or done something wrong.  Thanks!

marceldegraaf commented on lib/google_places/client.rb:
```diff
@@ -148,7 +148,40 @@ def spot(place_id, options = {})
+    # @option options [Integer] :retry_options[:max] (0) the maximum retries
+    # @option options [Integer] :retry_options[:delay] (5) the delay between each retry in seconds
+    #
+    # @see http://spreadsheets.google.com/pub?key=p9pdwsai2hDMsLkXsoM05KQ&gid=1 List of supported languages
```
When opening this link I get an error: "We're sorry, this document is not published". Can you fix the link to a working version?

TNCodeMonkey replied:
Thank you for the feedback.  I will knock those out very soon!  Your lib
was awesome for my application, just glad I could give back.

--Dustin
On Feb 25, 2015 5:27 AM, "Marcel de Graaf" <notifications@github.com> wrote:

> In lib/google_places/client.rb
> <https://github.com/marceldegraaf/google_places/pull/74#discussion_r25333218>
> :
>
> > +    # @option bounds [:se][String, Integer] :lng
> > +    #   the SE longitude
> > +    # @option options [String,Array] :query
> > +    #   Restricts the results to Spots matching term(s) in the specified query
> > +    # @option options [String] :language
> > +    #   The language code, indicating in which language the results should be returned, if possible.
> > +    # @option options [String,Array<String>] :exclude ([])
> > +    #   A String or an Array of <b>types</b> to exclude from results
> > +    #
> > +    # @option options [Hash] :retry_options ({})
> > +    #   A Hash containing parameters for search retries
> > +    # @option options [Object] :retry_options[:status] ([])
> > +    # @option options [Integer] :retry_options[:max] (0) the maximum retries
> > +    # @option options [Integer] :retry_options[:delay] (5) the delay between each retry in seconds
> > +    #
> > +    # @see http://spreadsheets.google.com/pub?key=p9pdwsai2hDMsLkXsoM05KQ&gid=1 List of supported languages
>
> When opening this link I get an error: "We're sorry, this document is not
> published". Can you fix the link to a working version?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/marceldegraaf/google_places/pull/74/files#r25333218>.
>

marceldegraaf replied:
Awesome, looking forward to it! :smile:

marceldegraaf commented on lib/google_places/rectangle.rb:
```diff
@@ -0,0 +1,19 @@
+
+    def format
+      #SW point of rectangle
+      a1 = [@lat1, @lat2].join (',')
```
I would add spaces inside the Array (e.g. `[ @lat1, @lat2 ]`) and remove the space in `join (',')`

marceldegraaf commented on lib/google_places/spot.rb:
```diff
@@ -1,5 +1,5 @@
 require 'google_places/review'
-
+require 'google_places'
```
Why did you add this `require`?

TNCodeMonkey replied:
We can chalk this up to being a Ruby noobie :P .  Pulling this out for next commit

marceldegraaf replied:
No problem! :+1:

marceldegraaf commented on lib/google_places/spot.rb:
```diff
@@ -82,7 +82,67 @@ def self.list(lat, lng, api_key, options = {})
+      # @see http://spreadsheets.google.com/pub?key=p9pdwsai2hDMsLkXsoM05KQ&gid=1 List of supported languages
+      # @see https://developers.google.com/maps/documentation/places/supported_types List of supported types
+    def self.list_by_bounds(bounds, api_key, options = {})
+      multipage_request = !!options.delete(:multipage)
```
Can you refactor `!!options...` to something a bit more verbose? It's hard to see what's going on here.

marceldegraaf commented on lib/google_places/spot.rb:
```diff
@@ -82,7 +82,67 @@ def self.list(lat, lng, api_key, options = {})
+    def self.list_by_bounds(bounds, api_key, options = {})
+      multipage_request = !!options.delete(:multipage)
+      rankby = options.delete(:rankby)
+      rect = Rectangle.new(bounds[:sw][:lat], bounds[:sw][:lng],bounds[:ne][:lat], bounds[:ne][:lng])
```
Missing a space between `,bounds[:ne][:lat]...`

marceldegraaf commented on lib/google_places/rectangle.rb:
```diff
@@ -0,0 +1,19 @@
+module GooglePlaces
+  class Rectangle
+    def initialize(lat1, lng1, lat2, lng2)
```
Isn't it more understandable to use something like `north_west`, `north_east`, `south_east`, and `south_west` here, to indicate the outer points of the rectangle? Latitude and longitude may be a bit too abstract?

TNCodeMonkey replied:
I wanted to leave the option option to create the rectangle from any start+end point combination -- NE + SW, SE + NW, NW + SE, and so on.  I was afraid that specifying SE/NW would infer to the user that they could only define bounds by that given pair.  Do you think start_lat, start_lng, end_lat, end_lng would gain us some clarity?

marceldegraaf replied:
Alright, if you can make sure to document (in a comment) sufficiently what these variables do and mean, it's fine with me :smile:

marceldegraaf commented:
Thanks @TNCodeMonkey, looks good, aside from a few formatting issues. If you can take a look at those, I'd be happy to merge this PR in and push out a new gem version.

TNCodeMonkey commented:
Ok @marceldegraaf , I think I've updated everything that needed attention.  Also, I went ahead and added test coverage for rectangle and Client.spots_by_bounds.  Please let me know if you need anything else taken care of with these changes.  Thanks!

marceldegraaf commented:
Awesome, thanks! Version 0.30.0 of the gem was just released, and it contains this addition. Nice work @TNCodeMonkey :smile:

TNCodeMonkey commented:
Thank you for your critique, actually made my first OSS contribution very painless :)  Look forward to adding more in the future!!

marceldegraaf commented:
No problem!
